src/main.rs

```diff
@@ -1,42 +1,33 @@
 use std::fmt::{Debug, Display};
-use std::fs::File;
+use std::sync::Arc;
 
+use tokio::sync::{broadcast, RwLock};
 use tokio::task::JoinError;
 
-use barcodequery::barcode_query::barcode_query_hashstorage_impl::BarCodeFileHashStorageImpl;
-use barcodequery::barcode_query::storage::BarCodeStorage;
-use barcodequery::barcode_query_app::app_async::BarcodeQueryAppAsync;
-use barcodequery::barcode_reader::console_reader::ConsoleBarcodeReader;
+use barcodequery::barcode_query::model::Message;
+use barcodequery::barcode_query_app::app_async::BarcodeAppAsync;
 use barcodequery::web::server_axum::WebSocketServer;
+use barcodequery::web::state::State;
 
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
-    // TODO add barcode app: receives barcode and query part
-    let server_task = tokio::spawn(WebSocketServer::run_until_stopped());
+    let (sender1, receiver) = broadcast::channel::<Message>(20);
+    let shared_state = Arc::new(RwLock::new(State {
```

Comment:
We should use `eventual fairness` instead of `fairness`. Using `parking_lot` instead of std https://docs.rs/parking_lot/latest/parking_lot/type.RwLock.html